src/containers/RapidAppointment/RapidAppointment.tsx

```diff
@@ -1,9 +1,603 @@
+  const [shouldShowInvalidURL, setShouldShowInvalidURL] = useState(false);
+
+  const [name, setName] = useState("");
+  const [address, setAddress] = useState("");
+  const [city, setCity] = useState("");
+  const [province, setProvince] = useState("");
+  const [postalCode, setPostalCode] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [website, setWebsite] = useState("");
+  const [numAvailable, setNumAvailable] = useState("");
+  const [vaccineTypeString, setVaccineTypeString] = useState(
+    "Select Vaccine Type",
+  );
+  const [vaccineId, setVaccineId] = useState(1);
+
+  const [isPopOverActive, setIsPopOverActive] = useState(false);
+  const [isExpiringDosesChecked, setIsExpiringDosesChecked] = useState(false);
+  const [isCancellationsChecked, setIsCancellationsChecked] = useState(false);
+  const [isCallAheadChecked, setIsCallAheadChecked] = useState(false);
+  const [isWalkInChecked, setIsWalkInChecked] = useState(false);
+  const [isVisitWebsiteChecked, setIsVisitWebsiteChecked] = useState(false);
+  const [isEmailChecked, setIsEmailChecked] = useState(false);
+  const [shouldShowExpandedForm, setShouldShowExpandedForm] = useState(true);
+
+  const previousData = usePrevious(data);
+  useEffect(() => {
+    if (!previousData && data) {
+      setName(data.name);
+      setAddress(data.address?.line1 || "");
+      setCity(data.address?.city || "");
+      setProvince(data.address?.province || "");
+      setPostalCode(data.address?.postcode || "");
+      setPhoneNumber(data.phone || "");
+      setWebsite(data.url || "");
+      setShouldShowExpandedForm(false);
+    }
+  }, [data, previousData]);
+  const validateForm = () => {
+    let isValid = true;
+    if (!name) {
+      setShouldShowInvalidName(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidName(false);
+    }
+
+    if (!address) {
+      setShouldShowInvalidAddress(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidAddress(false);
+    }
+
+    if (!city) {
+      setShouldShowInvalidCity(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidCity(false);
+    }
+
+    if (!province) {
+      setShouldShowInvalidProvince(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidProvince(false);
+    }
+
+    if (!postalCodeIsValid(postalCode)) {
+      setShouldShowInvalidPostal(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidPostal(false);
+    }
+
+    if (!website) {
+      setShouldShowInvalidURL(true);
+      isValid = false;
+    } else {
+      setShouldShowInvalidURL(false);
+    }
+    return isValid;
+  };
+
+  const handleSubmit = () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    let tagsCommaSeperatedString = "";
+
+    if (isCallAheadChecked) {
+      tagsCommaSeperatedString = "Call Ahead";
+    }
+    if (isWalkInChecked) {
+      if (!tagsCommaSeperatedString) {
```

Comment:
Instead of this, it would be better to add all tags to an array. When you're done, call `.join(",")` on the array to concatenate the values separated by commas.